## **Contexto:**

Son el equipo de Inteligencia Artificial de una firma de análisis móvil. Necesitan entrenar un modelo para predecir el éxito de una app, pero el archivo CSV que les mandó el equipo de Scraping es un desastre de formato. Tienen que convertirlo en una matriz matemática estricta.

In [6]:
import pandas as pd
import numpy as np
 
 
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/googleplaystore.csv"
df = pd.read_csv(url)

El scraper cometió errores y guardó la misma app varias veces. Escriban una línea de código para identificar si hay nombres duplicados en la columna App (df.duplicated(subset=['App']).sum()). Si los hay, apliquen .drop_duplicates(subset=['App'], keep='first', inplace=True) para quedarse solo con la primera aparición de cada aplicación.

## **Pregunta A: ¿Cuántos registros duplicados purgaron?**

In [7]:
print(df.duplicated(subset=['App']).sum())
df.drop_duplicates(subset=['App'], keep='first', inplace=True)

1181


A: 1181 duplicados.

La columna Installs tiene texto como "1,000,000+". Un algoritmo no puede procesar eso.

## **Pregunta B:**
### **a. Escriban una instrucción encadenando el método .str.replace() para eliminar el símbolo + y otro para eliminar la coma ,. (Ojo: utilicen regex=False o escapen el símbolo más \+).**

### **b. Al final de la cadena, utilicen .astype(int) (o float si tienen NaNs) para convertir la columna.**

### **c. Comprueben el éxito ejecutando df['Installs'].mean(). ¿Cuál es el promedio real de descargas?**

In [8]:
df['Installs'] = df['Installs'].str.replace('+', '', regex=False).str.replace(',', '', regex=False)
df = df[df['Installs'].str.isnumeric()]
df['Installs'] = df['Installs'].astype(int)
df['Installs'].mean()

np.float64(7777506.732270421)

B: Promedio de descargas: 7,777,507 aprox

La columna Price tiene valores como "$4.99" y "0". 

## **Pregunta C: Apliquen la misma lógica arquitectónica que en la Pregunta B: eliminen el símbolo de dólar $ y conviertan la columna usando .astype(float).**

In [9]:
df['Price'] = df['Price'].str.replace('$', '', regex=False).astype(float)
df['Price'].max()

400.0

C: despues de la conversion, el precio máximo encontrado es 400.0.

Ahora que el precio es numérico, ejecuten df['Price'].max(). Notarán que la app más cara cuesta alrededor de 400 dólares.

## **Pregunta D:**
### **a. Usen un filtro booleano (df[df['Price'] > 200]) para descubrir cuáles son estas apps. Verán que son aplicaciones basura o de broma.**

### **b. Si metemos estas apps al modelo, la red neuronal creerá que vender a 400 dólares es un modelo de negocio viable. Sobrescriban el DataFrame conservando únicamente las apps que cuesten menos de 50 dólares (df = df[df['Price'] < 50]).**

### **c. Guarden su dataset final usando df.to_csv('playstore_limpio.csv', index=False).**

In [10]:
print(df[df['Price'] > 200]['App'].tolist())
df = df[df['Price'] < 50]
df.to_csv('playstore_limpio.csv', index=False)

['most expensive app (H)', "💎 I'm rich", "I'm Rich - Trump Edition", 'I am rich', 'I am Rich Plus', 'I am rich VIP', 'I Am Rich Premium', 'I am extremely Rich', 'I am Rich!', 'I am rich(premium)', 'I Am Rich Pro', 'I am rich (Most expensive app)', 'I Am Rich', 'I am Rich', 'I AM RICH PRO PLUS', 'Eu Sou Rico', "I'm Rich/Eu sou Rico/أنا غني/我很有錢"]


D: efectivamente, son apps basura. Despues de filtrar "precio < 50" quedan 9636 apps.

In [11]:
df.describe()

,Rating,Installs,Price
count,8180.000000,9.636000e+03,9636.000000
mean,4.173753,7.796048e+06,0.336586
std,0.536736,5.382106e+07,1.899534
min,1.000000,0.000000e+00,0.000000
25%,4.000000,1.000000e+03,0.000000
50%,4.300000,1.000000e+05,0.000000
75%,4.500000,1.000000e+06,0.000000
max,5.000000,1.000000e+09,46.990000
